pkg/providers/vsphere/template.go

```diff
@@ -275,8 +278,20 @@ func buildTemplateMapCP(
 		values["etcdSshUsername"] = firstEtcdMachinesUser.Name
 		values["vsphereEtcdSshAuthorizedKey"] = etcdSSHKey
 
-		if etcdMachineSpec.HostOSConfiguration != nil && etcdMachineSpec.HostOSConfiguration.NTPConfiguration != nil {
-			values["etcdNtpServers"] = etcdMachineSpec.HostOSConfiguration.NTPConfiguration.Servers
+		if etcdMachineSpec.HostOSConfiguration != nil {
+			if etcdMachineSpec.HostOSConfiguration.NTPConfiguration != nil {
+				values["etcdNtpServers"] = etcdMachineSpec.HostOSConfiguration.NTPConfiguration.Servers
+			}
+
+			if etcdMachineSpec.HostOSConfiguration.BottlerocketConfiguration != nil {
+				if etcdMachineSpec.HostOSConfiguration.BottlerocketConfiguration.Kernel != nil &&
+					etcdMachineSpec.HostOSConfiguration.BottlerocketConfiguration.Kernel.SysctlSettings != nil {
+					values["etcdKernelSettings"], err = mapKernelSettingsToYaml(etcdMachineSpec.HostOSConfiguration.BottlerocketConfiguration.Kernel.SysctlSettings)
```

Comment:
instead of mapping this settings to yaml, can we handle the rendering in the template?